### PR TITLE
Add end-to-end integration test to assert correctness

### PR DESCRIPTION
**Pull request by xichen2020**
cc @jeromefroe @cw9 

This PR adds end-to-end integration test to assert functional correctness of the collector.

### CODE REVIEW

File: integration/report_with_rule_updates_test.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+func testReportWithRuleUpdates(
+	t *testing.T,
+	testDesc string,
+	intputIDGen idGenerator,
+	outputRes outputResults,
+	iterPool id.SortedTagIteratorPool,
+) {
+
+	namespaces := defaultNamespaces()
```
Comment (jeromefroe) [inline integration/report_with_rule_updates_test.go]: Would it be worthwhile to move these into the function signature? When looking at the integration tests, such as `TestReportMatchMappingWithRuleUpdates `, it's unclear from the test why the id's match a given mapping or rollup rule.

File: integration/data.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+type resultsByTypeAscIDAsc []metricWithPoliciesList
+
+func (r resultsByTypeAscIDAsc) Len() int      { return len(r) }
+func (r resultsByTypeAscIDAsc) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
+
+func (r resultsByTypeAscIDAsc) Less(i, j int) bool {
+	it, iid := metricTypeAndIDFrom(r[i].metric)
+	jt, jid := metricTypeAndIDFrom(r[j].metric)
+	if it < jt {
```
[the remark this answers is not in the document]
Comment (xichen2020) [reply integration/data.go]: Hehe I feel it's easier to understand if I explicitly spell out the conditions.

File: integration/report_with_rule_updates_test.go (hunk 1 above)
Comment (xichen2020) [reply integration/report_with_rule_updates_test.go]: Sure thing

**Comment by xichen2020** [comment]
Will add README as well

**Pull request merged**